components/datadog/apps/nginx/k8s/modifiers.go

```diff
@@ -0,0 +1,164 @@
+package k8s
+
+import (
+	appsv1 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/apps/v1"
+	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/core/v1"
+	metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/meta/v1"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+// DeploymentModifier is a function that operates on a DeploymentArgs struct
+type DeploymentModifier func(args *appsv1.DeploymentArgs)
+
+// rawMapToStringMap converts map[string]string
+func rawMapToStringMap(raw map[string]string) pulumi.StringMap {
+	result := pulumi.StringMap{}
+	for k, v := range raw {
+		result[k] = pulumi.String(v)
+	}
+	return result
+}
```

Comment:
Such a fonction already exists.
It’s [`ToStringMap`](https://github.com/pulumi/pulumi/blob/24c16fc0f22fd6aaf428887ceb4ea015f2be4ea9/sdk/go/pulumi/types_builtins.go#L5215-L5221).

```suggestion
```